File: core/src/main/java/com/palomiklo/streampulse/constant/Default.java
package com.palomiklo.streampulse.constant;

import java.util.function.Supplier;

public interface Default {

    byte CONNECTION_TIMEOUT = 1 * 60;
    byte INITIAL_DELAY = 0;
    byte PING_INTERVAL = 5;
    short CONNECTION_CLEAN_UP_TIMEOUT = 5 * 60;
    Supplier<String> PING = () -> "ping";
}

File: core/src/main/kotlin/com/palomiklo/streampulse/constant/Default.kt
package com.palomiklo.streampulse.constant

import java.util.function.Supplier

interface Default {
    companion object {
        const val CONNECTION_TIMEOUT: Byte = (1 * 60).toByte()
        const val INITIAL_DELAY: Byte = 0
        const val PING_INTERVAL: Byte = 5
        const val CONNECTION_CLEAN_UP_TIMEOUT: Short = (5 * 60).toShort()
        val PING: Supplier<String> = Supplier<String> { "ping" }
        val RECONNECT_EVENT: Supplier<String> = Supplier<String> { "reconnect" }
    }
}
